scripts/utils/build.js

```diff
@@ -54,38 +57,111 @@ function prefixPlugin (prefixMessage) {
+ * @param {string} [params.platform]
+ * @param {string[]} [params.featureNames]
+ * @param {string} [params.name]
+ * @param {boolean} [params.supportsMozProxies]
+ * @return {Promise<string>}
+ */
+export async function rollupScript (params) {
+    const {
+        scriptPath,
+        platform,
+        name,
+        featureNames,
+        supportsMozProxies = false
+    } = params
+
     // The code is using a global, that we define here which means once tree shaken we get a browser specific output.
     const mozProxies = supportsMozProxies
 
-    const inputOptions = {
-        input: scriptPath,
-        plugins: [
-            runtimeInjections(),
-            resolve(),
-            dynamicImportVariables({}),
-            commonjs(),
-            replace({
-                preventAssignment: true,
-                values: {
-                    // @ts-expect-error https://app.asana.com/0/1201614831475344/1203979574128023/f
-                    mozProxies
-                }
-            }),
-            prefixPlugin(prefixMessage)
-        ]
+    const plugins = [
+        runtimeInjections(),
+        resolve(),
+        commonjs(),
+        replace({
+            preventAssignment: true,
+            values: {
+                // @ts-expect-error https://app.asana.com/0/1201614831475344/1203979574128023/f
+                mozProxies,
+                'import.meta.injectName': JSON.stringify(platform)
+            }
+        }),
+        prefixPlugin(prefixMessage)
+    ]
+
+    // if 'platform' was provided, ensure we include the dyn
+    if (platform) {
+        plugins.unshift(loadFeatures(platform, featureNames))
     }
-    const outputOptions = {
+
+    const bundle = await rollup.rollup({
+        input: scriptPath,
+        plugins
+    })
+
+    const generated = await bundle.generate({
         dir: 'build',
         format: 'iife',
         inlineDynamicImports: true,
         name,
         // This if for seedrandom causing build issues
         globals: { crypto: 'undefined' }
-    }
+    })
 
-    const bundle = await rollup.rollup(inputOptions)
-    // @ts-expect-error https://app.asana.com/0/1201614831475344/1203979574128023/f
-    const generated = await bundle.generate(outputOptions)
     return generated.output[0].code
 }
+
+/**
+ * Include the features required on each platform
+ *
+ * @param {string} platform
+ * @param {string[]} featureNames
+ */
+function loadFeatures (platform, featureNames = platformSupport[platform]) {
+    const pluginId = 'ddg:platformFeatures'
+    return {
+        name: pluginId,
+        resolveId (id) {
+            if (id === pluginId) return id
+            return null
+        },
+        load (id) {
+            if (id !== pluginId) return null
+
+            // convert a list of feature names to
+            const imports = featureNames.map((f, index) => {
+                const fileName = f.replace(/([a-zA-Z])(?=[A-Z0-9])/g, '$1-').toLowerCase()
```

Comment:
Can we share this regex with the `getFilename`?

Reply:
oh yeah, that was the point of moving it haha, missed it